src/AdoPat/PatClient.cs

```diff
@@ -0,0 +1,139 @@
+        /// Initializes a new instance of the <see cref="PatClient"/> class.
+        /// </summary>
+        /// <param name="client">Any class which implements the <see cref="ITokensHttpClientWrapper"/> interface.</param>
+        public PatClient(ITokensHttpClientWrapper client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Creates a new PAT.
+        /// </summary>
+        /// <param name="patTokenCreateRequest">
+        /// XXX: Not documented by Visual Studio client libraries. Should we depend on this or
+        /// should this method use the corresponding primitives as arguments? Not yet clear.
+        /// </param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatTokenResult"/>.</returns>
+        public async Task<PatToken> CreatePatAsync(
+            PatTokenCreateRequest patTokenCreateRequest,
+            CancellationToken cancellationToken = default)
+        {
+            var patTokenResult = await this.client.CreatePatAsync(
+                patTokenCreateRequest,
+                cancellationToken: cancellationToken)
+            .ConfigureAwait(false);
+
+            if (patTokenResult.PatToken == null)
+            {
+                throw new PatClientException($"Failed to create PAT during regeneration: {patTokenResult.PatTokenError}");
+            }
+
+            return patTokenResult.PatToken;
+        }
+
+        /// <summary>
+        /// Gets all active PATs.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>The set of active PATs.</returns>
+        public async Task<ISet<PatToken>> GetActivePatsAsync(CancellationToken cancellationToken = default)
+        {
+            // Initialize a PagedPatTokens so that we can use the continuation token
+            // in the scope of the conditional of the do while loop below. Azure
+            // DevOps will return the empty string when there are no more pages.
+            var pagedPatTokens = new PagedPatTokens(continuationToken: string.Empty, patTokens: new List<PatToken>());
+            var pats = new HashSet<PatToken>();
+            do
+            {
+                pagedPatTokens = await this.client.ListPatsAsync(
+                    displayFilterOption: DisplayFilterOptions.Active,
+                    top: PageSize,
+                    continuationToken: pagedPatTokens.ContinuationToken,
+                    sortByOption: SortByOptions.DisplayDate,
+                    cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+
+                foreach (PatToken pat in pagedPatTokens.PatTokens)
+                {
+                    pats.Add(pat);
+                }
+            }
+            while (!string.IsNullOrEmpty(pagedPatTokens.ContinuationToken));

```

Comment:
Is there a benefit of using `do-while` loop versus a simple `while` loop here?

Reply:
Hmm... I think I tried to use a `do...while` here, but couldn't make it work the way I wanted. If you can demonstrate that working here I'd be happy to switch to more terse code, @shalinikhare27! ❤️